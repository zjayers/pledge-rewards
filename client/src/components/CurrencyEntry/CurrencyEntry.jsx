import React from 'react';
import { CurrencyBox, CurrencyIcon, Form, Input } from './CurrencyEntry.style';

const CurrencyEntry = (props) => {
  const { input, setInput } = props;

  const handleChange = (e) => {
    setInput(e.target.value);
  };

  return (
    <Form>
      <Input type='number' min='0' value={input} onChange={handleChange} />
      <CurrencyBox>
        <CurrencyIcon>$</CurrencyIcon>
      </CurrencyBox>
    </Form>
  );
};

export default CurrencyEntry;
