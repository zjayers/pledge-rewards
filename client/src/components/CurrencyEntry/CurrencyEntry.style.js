import styled from 'styled-components';

export const Form = styled.div`
  position: relative;
  margin: 10px 0;
`;

export const CurrencyIcon = styled.span`
  justify-content: center;
  align-items: center;
  text-align: center;
  overflow: hidden;
  font-size: 1.2rem;
  white-space: nowrap;
  width: 4.2rem;
`;

export const CurrencyBox = styled.div`
  transition: all 300ms cubic-bezier(0.175, 0.885, 0.335, 1);
  height: 100%;
  position: absolute;
  border-right: 1px solid #dcdedd;
  top: 0;
  left: 0;
  display: inline-flex;
  align-items: center;
  pointer-events: none;
  color: #656969;
  z-index: 2;
`;

export const Input = styled.input`
  border: 1px solid #dcdedd;
  color: #282828;
  cursor: default;
  padding: 10px 10px 10px 5.4rem;
  position: relative;
  height: 42px;
  line-height: 20px;
  box-sizing: border-box;
  width: 100%;
  transition: all 300ms cubic-bezier(0.175, 0.885, 0.335, 1);
  text-rendering: auto;
  word-spacing: normal;
  text-transform: none;
  text-indent: 0px;
  text-shadow: none;
  display: inline-block;
  text-align: start;
  font: 400 13.3333px Arial;
  writing-mode: horizontal-tb;

  &:focus {
    outline: 0;
    border-color: #009e74;
    box-shadow: #f8f9fd 0 0 5px;

    & + div {
      border-color: #009e74;
    }
  }
`;
