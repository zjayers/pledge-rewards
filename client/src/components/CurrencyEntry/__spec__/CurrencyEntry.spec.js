import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CurrencyEntry from '../CurrencyEntry';
import { Input } from '../CurrencyEntry.style';

describe('CurrencyEntry Component', () => {
  let component;
  let handleChangeMock;

  beforeEach(() => {
    handleChangeMock = jest.fn();
    component = mount(<CurrencyEntry input={0} setInput={handleChangeMock} />);
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should allow the user to enter numbers into the text box', () => {
    component.find(Input).simulate('change', { target: { value: 1 } });
    expect(handleChangeMock).toHaveBeenCalledTimes(1);
  });
});
