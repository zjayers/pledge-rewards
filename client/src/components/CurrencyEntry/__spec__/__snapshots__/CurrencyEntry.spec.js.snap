// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyEntry Component should match the test snapshot 1`] = `
<CurrencyEntry
  input={0}
  setInput={[MockFunction]}
>
  <CurrencyEntrystyle__Form>
    <div
      className="CurrencyEntrystyle__Form-zl3z4f-0 cCvtGG"
    >
      <CurrencyEntrystyle__Input
        min="0"
        onChange={[Function]}
        type="number"
        value={0}
      >
        <input
          className="CurrencyEntrystyle__Input-zl3z4f-3 kOuXyP"
          min="0"
          onChange={[Function]}
          type="number"
          value={0}
        />
      </CurrencyEntrystyle__Input>
      <CurrencyEntrystyle__CurrencyBox>
        <div
          className="CurrencyEntrystyle__CurrencyBox-zl3z4f-2 jTmnGB"
        >
          <CurrencyEntrystyle__CurrencyIcon>
            <span
              className="CurrencyEntrystyle__CurrencyIcon-zl3z4f-1 evsVHR"
            >
              $
            </span>
          </CurrencyEntrystyle__CurrencyIcon>
        </div>
      </CurrencyEntrystyle__CurrencyBox>
    </div>
  </CurrencyEntrystyle__Form>
</CurrencyEntry>
`;
